src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package mate.service;
+
+import java.util.Optional;
+import mate.exception.AuthenticationException;
+import mate.exception.DataProcessingException;
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String username, String password) throws AuthenticationException {
+        Optional<Driver> driver;
+        try {
+            driver = driverService.findByLogin(username);
+        } catch (DataProcessingException e) {
```

Comment:
let's use single if construction in this method, plus you don't need to have try catch. If code is working fine you won't have DataProcessingException. But if it happens it means SQL code is wrong and we need to fix it